src/BlockTypes/CollectionFilters.php

```diff
@@ -0,0 +1,179 @@
+ */
+class CollectionFilters extends AbstractBlock {
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name = 'collection-filters';
+
+	/**
+	 * Mapping inner blocks to CollectionData API parameters.
+	 *
+	 * @var array
+	 */
+	protected $collection_data_params_mapping = array(
+		'calculate_price_range'         => 'woocommerce/collection-price-filter',
+		'calculate_stock_status_counts' => 'woocommerce/collection-stock-filter',
+		'calculate_attribute_counts'    => 'woocommerce/collection-attribute-filter',
+		'calculate_rating_counts'       => 'woocommerce/collection-rating-filter',
+	);
+
+	/**
+	 * Cache the current response from the API.
+	 *
+	 * @var array
+	 */
+	protected $current_response = null;
+
+	/**
+	 * Get the frontend style handle for this block type.
+	 *
+	 * @return null
+	 */
+	protected function get_block_type_style() {
+		return null;
+	}
+
+	/**
+	 * Get the frontend script handle for this block type.
+	 *
+	 * @param string $key Data to get, or default to everything.
+	 *
+	 * @return null This block has no frontend script.
+	 */
+	protected function get_block_type_script( $key = null ) {
+		return null;
+	}
+
+	/**
+	 * Initialize this block type.
+	 *
+	 * - Hook into WP lifecycle.
+	 * - Register the block with WordPress.
+	 */
+	protected function initialize() {
+		parent::initialize();
+		add_action( 'render_block_context', array( $this, 'modify_inner_blocks_context' ), 10, 3 );
+	}
+
+	/**
+	 * Extra data passed through from server to client for block.
+	 *
+	 * @param array $attributes  Any attributes that currently are available from the block.
+	 *                           Note, this will be empty in the editor context when the block is
+	 *                           not in the post content on editor load.
+	 */
+	protected function enqueue_data( array $attributes = [] ) {
+		parent::enqueue_data( $attributes );
+
+		if ( ! is_admin() ) {
+			/**
+			 * At this point, WP starts rendering the Collection Filters block,
+			 * we can safely nuke the current response.
+			 */
+			$this->current_response = null;
+		}
+	}
+
+	/**
+	 * Modify the context of inner blocks.
+	 *
+	 * @param array    $context The block context.
+	 * @param array    $parsed_block The parsed block.
+	 * @param WP_Block $parent_block The parent block.
+	 * @return array
+	 */
+	public function modify_inner_blocks_context( $context, $parsed_block, $parent_block ) {
+		if (
+			is_admin() ||
+			! is_a( $parent_block, 'WP_Block' ) ||
+			"woocommerce/{$this->block_name}" !== $parent_block->name ||
+			empty( $parent_block->inner_blocks )
+		) {
+			return $context;
+		}
+
+		if ( ! isset( $this->current_response ) ) {
+			$this->current_response = $this->get_aggregated_collection_data( $parent_block );
+		}
+
+		if (
+			! empty(
+				$this->current_response ||
+				isset( $parsed_block['blockName'] ) ||
+				in_array( $parsed_block['blockName'], $this->collection_data_params_mapping, true )
+			) ) {
```

Comment:
It seems to me that you wanted to do:

```suggestion
			! empty( $this->current_response ) ||
			isset( $parsed_block['blockName'] ) ||
			in_array( $parsed_block['blockName'], $this->collection_data_params_mapping, true )
			) ) {
```

Right?

Reply:
Yeah, I fixed this issue in https://github.com/woocommerce/woocommerce-blocks/pull/10919/files#diff-b01c934c955dd9083e3c6f80952eee42fd4abe0edaad735b3fdb398f89b64d4fR118-R120 but forgot to update this PR.

Reply:
@dinhtungdu should I pre-approve this PR then?

Reply:
@sunyatasattva No need, this one is the foundation for contextual and help feeding data to inner filter blocks it's better to review it carefully!